app/helpers/application_helper.rb

```diff
@@ -540,4 +540,22 @@ def replacement_event_notice(event_schedule)
   def quantity_left_of(resource)
     "#{resource.quantity - resource.used}/#{resource.quantity}"
   end
+
+  def concurrent_events(event)
+    return nil unless event.scheduled? && event.program.selected_event_schedules
+    event_schedule = event.program.selected_event_schedules.find_by(event: event)
+    other_event_schedules = event.program.selected_event_schedules.reject { |other_event_schedule| other_event_schedule == event_schedule }
+    concurrent_events = []
+
+    event_time_range = (event_schedule.start_time.strftime '%Y-%m-%d %H:%M')...(event_schedule.end_time.strftime '%Y-%m-%d %H:%M')
```

Comment:
I am using `.strftime` to remove the seconds part of the time. Suppose for an one hour event the number of elements in the array would be 60 each corresponding to each minute in event_time_range and other_event_time_range . Using `.to_datetime`  the number of elements would be 3600 in both arrays which will increase the number of comparisons made in line 554 . :sweat_smile: